File: Exercism Codes/JavaScript/saddle-points.js
export const saddlePoints = (matrix) => {
  if(matrix.length == 0) return [];
  const max = matrix.map(x => Math.max(...x));
  const min = [...Array(matrix[0].length).keys()]
    .map(x => Math.min(...matrix.map(y => y[x])));
  return [...Array(matrix.length).keys()]
    .flatMap(x => [...Array(matrix[x].length).keys()]
    .filter(y => matrix[x][y] >= max[x] && matrix[x][y] <= min[y])
    .map(y => ({ row: x + 1, column: y + 1 })));
};

File: Exercism Codes/Python/saddle_points.py
def saddle_points(matrix):
    if len(matrix) == 0:
        return {}
    try:
        maximum = [max(x) for x in matrix]
        minimum = [min(y[x] for y in matrix) for x in range(len(matrix[0]))]
        return flatten(
            [[{"row": x + 1, "column": y + 1} for y in range(len(matrix[x])) if maximum[x] <= matrix[x][y] <= minimum[y]]
             for x in range(len(matrix))])
    except:
        raise ValueError("irregular matrix")


def flatten(lst):
    return [i for item in lst for i in item]
